packages/subgraph/src/utils/tokens/erc20.ts

```diff
@@ -1,11 +1,61 @@
-import {Address, BigInt, Bytes, ethereum} from '@graphprotocol/graph-ts';
+import {Address, BigInt, Bytes, Value, ethereum} from '@graphprotocol/graph-ts';
 import {
   ERC20Balance,
   ERC20Contract,
-  ERC20Transfer
+  ERC20Transfer,
+  ERC20WrapperContract
 } from '../../../generated/schema';
 import {ERC20} from '../../../generated/templates/DaoTemplateV1_0_0/ERC20';
+import {GovernanceWrappedERC20} from '../../../generated/templates/GovernanceWrappedERC20/GovernanceWrappedERC20';
 import {ERC20_transfer, ERC20_transferFrom, getTransferId} from './common';
+import {supportsInterface} from '../erc165';
+import { WRAPPED_ERC20_INTERFACE } from '../constants';
+
+export function supportsERC20Wrapped(token: Address): bool {
+  // Double check that it's ERC20Wrapped by calling supportsInterface checks.
+  let erc20Wrapped = GovernanceWrappedERC20.bind(token);
+  let introspection_wrapped_erc20 = supportsInterface(erc20Wrapped, WRAPPED_ERC20_INTERFACE); // GovernanceWrappedERC20
+  let introspection_00000000 = supportsInterface(
+    erc20Wrapped,
+    '00000000',
+    false
+  );
+  return introspection_wrapped_erc20 && introspection_00000000;
```

Comment:
Why don't we simply return `false` as soon as we detect that `introspection_wrapped_erc20` is false? No need to keep computing/calling